From 567f180697ad7e4208dbda11578a923e48f222f1 Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Wed, 21 Sep 2016 10:37:44 +0200
Subject: [PATCH 397/595] Bug 771268 - [WK2] Prefer text over image on Paste

Prefer text and HTML content over others when pasting the clipboard content.
---
 composer/e-msg-composer.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/composer/e-msg-composer.c b/composer/e-msg-composer.c
index 23c29a2326..a2a9b89323 100644
--- a/composer/e-msg-composer.c
+++ b/composer/e-msg-composer.c
@@ -1720,17 +1720,25 @@ msg_composer_paste_clipboard_targets_cb (GtkClipboard *clipboard,
 	if (targets == NULL || n_targets < 0)
 		return;
 
+	editor = e_msg_composer_get_editor (composer);
+	cnt_editor = e_html_editor_get_content_editor (editor);
+
 	/* Order is important here to ensure common use cases are
 	 * handled correctly.  See GNOME bug #603715 for details. */
+	if (gtk_targets_include_text (targets, n_targets) ||
+	    e_targets_include_html (targets, n_targets)) {
+		if (composer->priv->last_signal_was_paste_primary) {
+			e_content_editor_paste_primary (cnt_editor);
+		} else
+			e_content_editor_paste (cnt_editor);
+		return;
+	}
 
 	if (gtk_targets_include_uri (targets, n_targets)) {
 		e_composer_paste_uris (composer, clipboard);
 		return;
 	}
 
-	editor = e_msg_composer_get_editor (composer);
-	cnt_editor = e_html_editor_get_content_editor (editor);
-
 	if (!e_content_editor_get_html_mode (cnt_editor) &&
 	    gtk_targets_include_image (targets, n_targets, TRUE)) {
 		e_composer_paste_image (composer, clipboard);
-- 
2.15.0

